app/src/main/java/com/keylesspalace/tusky/fragment/NotificationsFragment.java

```diff
@@ -255,10 +265,30 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
 
         ((SimpleItemAnimator) recyclerView.getItemAnimator()).setSupportsChangeAnimations(false);
 
+        updateFilterVisibility();
+
         return rootView;
     }
 
-    private void confirmClearNotifications(){
+    private void updateFilterVisibility() {
+        CoordinatorLayout.LayoutParams params =
+                (CoordinatorLayout.LayoutParams) swipeRefreshLayout.getLayoutParams();
+        if (showNotificationsFilter) {
+            appBarOptions.setExpanded(true, false);
+            appBarOptions.setVisibility(View.VISIBLE);
+            //Set content behaviour to hide filter on scroll
+            params.setBehavior(new AppBarLayout.ScrollingViewBehavior());
+        } else {
+            appBarOptions.setExpanded(false, false);
+            appBarOptions.setVisibility(View.GONE);
+            //Clear behaviour to hide app bar
+            params.setBehavior(null);
+        }
+
+        swipeRefreshLayout.requestLayout();
```

Comment:
this should not be necessary

Reply:
This is a workaround. If i hide the app bar without this workaround - it left the empty place instead. 
The same issue there: https://stackoverflow.com/questions/32704775/appbarlayout-take-space-after-setvisibilityview-gone/35126429

Reply:
This is a workaround. If i hide the app bar without this workaround - it leave the empty place instead of remove the bar.
https://stackoverflow.com/questions/32704775/appbarlayout-take-space-after-setvisibilityview-gone/35126429

Reply:
Yes, the setting of the scrolling behavior is necessary, but the `requestLayout` should not be.

Reply:
Ah. Sorry. Done.